core/src/test/java/org/kiwix/kiwixmobile/core/search/viewmodel/effects/OpenSearchItemTest.kt

```diff
@@ -39,8 +40,26 @@ internal class OpenSearchItemTest {
     val intent = mockk<Intent>()
     every {
       anyConstructed<Intent>().putExtra(TAG_FILE_SEARCHED, searchListItem.value)
+        .putExtra(TAG_FILE_SEARCHED_NEW_TAB, false)
     } returns intent
-    OpenSearchItem(searchListItem).invokeWith(activity)
+    OpenSearchItem(searchListItem, false).invokeWith(activity)
+    verify {
+      activity.setResult(Activity.RESULT_OK, intent)
+      activity.finish()
+    }
+  }
+
+  @Test
+  fun `invoke with returns an Ok Result with list item value for new tab`() {
```

Comment:
I don't have much experience with unit testing, especially not with mocking and in Kotlin. Is this a sufficient test, or is it necessary to check whether a new tab was actually created?